crates/fuel-core/src/query/message.rs

```diff
@@ -337,3 +341,184 @@ pub fn message_status(
+        }
+
+        fn insert_receipts(&mut self, transaction_id: TxId, receipts: Vec<Receipt>) {
+            self.receipts.insert(transaction_id, receipts);
+        }
+    }
+
+    impl MessageProofData for FakeDB {
+        fn block(&self, id: &BlockHeight) -> fuel_core_storage::Result<CompressedBlock> {
+            self.blocks.get(id).cloned().ok_or(not_found!("Block"))
+        }
+
+        fn transaction_status(
+            &self,
+            transaction_id: &TxId,
+        ) -> fuel_core_storage::Result<TransactionStatus> {
+            self.transaction_statuses
+                .get(transaction_id)
+                .cloned()
+                .ok_or(not_found!("Transaction status"))
+        }
+
+        fn block_history_proof(
+            &self,
+            _message_block_height: &BlockHeight,
+            _commit_block_height: &BlockHeight,
+        ) -> fuel_core_storage::Result<MerkleProof> {
+            // Unused in current tests
+            Ok(MerkleProof::default())
+        }
+
+        fn receipts(
+            &self,
+            transaction_id: &TxId,
+        ) -> fuel_core_storage::Result<Vec<Receipt>> {
+            self.receipts
+                .get(transaction_id)
+                .cloned()
+                .ok_or(not_found!("Receipts"))
+        }
+    }
+
+    // Test will try to get the message receipt proof with a block with only valid transactions
+    // Then add an invalid transaction and check if the proof is still the same (meaning the invalid transaction was ignored)
+    #[test]
+    fn test_message_receipts_proof_ignore_failed() {
+        // Create a fake database
+        let mut database = FakeDB::new();
+
+        // Given
+        // Create a block with a valid transaction and receipts
+        let mut block = CompressedBlock::default();
+        let valid_tx_id = Bytes32::new([1; 32]);
+        let mut valid_tx_receipts = vec![];
+        for i in 0..100 {
+            valid_tx_receipts.push(Receipt::MessageOut {
+                sender: Address::default(),
+                recipient: Address::default(),
+                amount: 0,
+                nonce: 0.into(),
+                len: 32,
+                digest: Bytes32::default(),
+                data: Some(vec![i; 32]),
+            });
+        }
+        block.transactions_mut().push(valid_tx_id);
+        database.insert_block(1u32.into(), block.clone());
+        database.insert_transaction_status(
+            valid_tx_id,
+            TransactionStatus::Success {
+                time: Tai64::UNIX_EPOCH,
+                block_height: 1u32.into(),
+                receipts: valid_tx_receipts.clone(),
+                total_fee: 0,
+                total_gas: 0,
+                result: None,
+            },
+        );
+        database.insert_receipts(valid_tx_id, valid_tx_receipts.clone());
+
+        // Get the message proof with the valid transaction
+        let message_proof_valid_tx = message_receipts_proof(
```

Comment:
It would be nice if we tested public function `message_proof` instead of the private

Reply:
Done